code/game/machinery/atmoalter/canister.dm

```diff
@@ -181,21 +182,27 @@ update_flag
 
 /obj/machinery/portable_atmospherics/canister/proc/healthcheck()
 	if(destroyed)
-		return 1
+		return TRUE
 
 	if (src.health <= 10)
 		var/atom/location = src.loc
 		location.assume_air(air_contents)
 
-		src.destroyed = 1
+		destroyed = TRUE
+		obj_flags &= ~OBJ_FLAG_SIGNALER
 		playsound(src.loc, 'sound/effects/spray.ogg', 10, 1, -3)
-		src.density = 0
-		update_icon()
+		density = FALSE
 
 		if (src.holding)
 			src.holding.forceMove(src.loc)
 			src.holding = null
 
+		if(signaler)
+			signaler.forceMove(get_turf(src))
+			signaler = null
```

Comment:
Do we need to nullify `signaler.machine` here too? Might be best to make some generic signaller machine detachment proc since do this in multiple places, like `/obj/machinery/attackby`